src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValuesTests.cs

```diff
@@ -425,6 +538,50 @@ End Enum
                 GetBasicResultAt(9, 5, "Value2", "1", "Value1"));
         }
 
+        [Fact]
+        public async Task EnumMemberBadConstantValue_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class C
+{
+    public static int I = 1;
+}
+
+public enum MyEnum
+{
+    Value1 = C.I
+}",
+                DiagnosticResult.CompilerError("CS0133").WithSpan(9, 14, 9, 17).WithArguments("MyEnum.Value1"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Module M
+    Public I As Integer = 1
+End Module
+
+Public Enum MyEnum
+    Value1 = M.I
+End Enum
+",
+                DiagnosticResult.CompilerError("BC30059").WithSpan(7, 14, 7, 17));
+        }
+
+        [Fact]
+        public async Task EnumMemberNullConstantValue_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public enum MyEnum
+{
+    Value1 = null
+}",
+                DiagnosticResult.CompilerError("CS0037").WithSpan(4, 14, 4, 18).WithArguments("int"));
```

Comment:
💡 For this case I would normally just use markup syntax `{|CS0037:...|}` since it's an expected part of the input.